core/src/bundle/usCoreBundleContext.cpp

```diff
@@ -59,19 +59,19 @@ std::map<std::string, Any> InitProperties(std::map<std::string, Any> configurati
   return configuration;
 }
 
-CoreBundleContext::CoreBundleContext(const std::map<std::string, Any>& props)
+CoreBundleContext::CoreBundleContext(const std::map<std::string, Any>& props, std::ostream* logger)
   : id(globalId++)
+  , frameworkProperties(InitProperties(props))
+  , sink(std::make_shared<LogSink>(logger, any_cast<bool>(frameworkProperties.at(Constants::FRAMEWORK_LOG))))
```

Comment:
I am not sure if a `shared_ptr` is worth it. We are not really sharing ownership of the log sink. Maybe a `unique_ptr` would be a better fit (and internally handing out `LogSink*`)?

Also the `any_cast<bool>` could throw an exception, preventing the framework from being constructed. Is this intended? If the property is not of type `bool`, I think we should fall back to using the default instead.